ci/scripts/autobump-dependencies.py

```diff
@@ -356,6 +356,28 @@ def fetch_latest_release(self) -> Release:
             version.parse(latest_version),
         )
 
+    # Parse and split version into parts removing the -dev suffix
+    def _parse_and_split_version(self, version):
+        if version.endswith('-dev'):
+            version = version[:-4]
+        version_parts = version.split('.')
+        return version_parts
+
+    # Returns normalized versions for changelog comparison
+    def _get_comparison_versions(self, current_version, latest_version):
+        comparison_current = str(current_version)
+        comparison_latest = str(latest_version)
+
+        current_parts = self._parse_and_split_version(comparison_current)
+        latest_parts = self._parse_and_split_version(comparison_latest)
```

Comment:
you could use decomposition here:


```suggestion
        current_major, current_minor, current_patch = self._parse_and_split_version(comparison_current)
        latest_major, latest_minor, latest_patch = self._parse_and_split_version(comparison_latest)
```

This forces that there are all 3 segments available always. They are and should always be anyway. If HAProxy changes their changelog format, we need to adjust this script anyway.

Then the comparison becomes:
```python
    if latest_major > current_major:
        comparison_current = f"{current_major}.{current_minor}.0"
```